eng/testing/linker/SupportFiles/Directory.Build.props

```diff
@@ -1,4 +1,13 @@
 <Project>
+
+  <!-- Use live illink bits. It is necessary to both import the package props and override
+       the tasks assembly, because the live package props in the build output do not use
+       the same layout as the NuGet package. -->
+  <Import Project="$(ToolsILLinkDir)build/Microsoft.NET.ILLink.Tasks.props" />
+  <PropertyGroup>
+    <ILLinkTasksAssembly>$(ToolsILLinkDir)net8.0/ILLink.Tasks.dll</ILLinkTasksAssembly>
```

Comment:
Can we use CurrentTargetFramework here? So we don’t have to update net8.0 in net 9 time?

Reply:
What is `CurrentTargetFramework`? I can't find that. I ask because I have an open PR with a similar change - https://github.com/dotnet/runtime/pull/89699 .

Reply:
https://github.com/dotnet/runtime/blob/30db9b78c8c21aea74a945332bda649e14b6b52d/src/libraries/Microsoft.Extensions.Caching.Abstractions/src/Microsoft.Extensions.Caching.Abstractions.csproj#L4

I meant NetCoreAppCurrent

Reply:
I think this should use the `NetCoreAppToolCurrent` because that's what the target project uses: https://github.com/dotnet/runtime/blob/242242947c01615b243e8b4d9ee163030a4ceb47/src/tools/illink/src/ILLink.Tasks/ILLink.Tasks.csproj#L4

Reply:
Note that `$(TargetFramework)` shouldn't be used in a props file as that property isn't defined at that time for non multi-targeting projects. You could move this property into the Directory.Build.targets file.

Reply:
Alternatively, you could add the TFM as a property to the imported props file, to keep them in sync.

Reply:
I wonder if we should make tools use the runtime configuration. For example when compiling AOT, the "tools" (ilc compiler) use runtime configuration. Additionally I ran into this myself and found it weird that specifying all of `-rc`, `-lc` and `-hc` wasn't enough, that there were still parts which didn't follow either of them.
(Obviously as a separate change)

Reply:
I asked about that when we brought the linker into runtime and the consensus was that the linker should not follow the runtime configuration, as its a separate tool but doesn't need to have yet another configuration property. The current configuration matrix is already quite complicated in dotnet/runtime.

Reply:
Do we have other code which falls into the same bucket (as not belonging into any of the 3 configuration properties)?

Reply:
Here are the current configuration properties: https://github.com/dotnet/runtime/blob/11d9e87db295032a9927ee19bf3e2918539d8941/eng/Subsets.props#L548-L553

> Do we have other code which falls into the same bucket (as not belonging into any of the 3 configuration properties)?

The shared framework package projects: "packs". But those don't invoke the compiler.

Reply:
> I don't think this works because the imported file ships with the package (so needs to hard code the package TFM) - unless I misunderstood the suggestion.

You already hardcode the TFM in the shipping props file: https://github.com/dotnet/runtime/blob/e43cba31e997e960205844b3f0541011ef732c29/src/tools/illink/src/ILLink.Tasks/build/Microsoft.NET.ILLink.Tasks.props#L21-L22

I was saying that you could introduce another property in that file that just includes the `net8.0\ILLink.Tasks.dll` part that you could then use here to keep the TFM in sync.